python/lsst/analysis/tools/actions/scalar/scalarActions.py

```diff
@@ -67,8 +67,8 @@ class MedianAction(ScalarFromVectorAction):
 
     def __call__(self, data: KeyedData, **kwargs) -> Scalar:
         mask = self.getMask(**kwargs)
-        values = data[self.vectorKey.format(**kwargs)][mask]
-        med = nanMedian(values) if len(values) else np.NaN
+        values = np.array(data[self.vectorKey.format(**kwargs)])[mask]
```

Comment:
Ok, so this is going to seem a bit weird, as all worrying about typing is in python. As of now Vector and Tensor are two different things, and actually have two different interfaces. As such we really should be checking for each. For instance:
```
values = np.from_dlpack(info) if isinstance((info := data[self.vectorKey.format(**kwargs)]), Tensor) else np.array(info)
values = values[mask]
```
The reason to do this is things that satisfy the Tensor protocol can be turned into numpy arrays with zero copies, but not everything that satisfies the Vector interface satisfy the Tensor interface, such as pandas sequences. (In this case it turns out they can be turned into arrays with zero copy, but as a quirk with an old style numpy interface).
We can of course support more restrictive interfaces, but up to this point we have been trying hard to support whatever data someone might pass in.

Reply:
Thanks. I've made the suggested change by adding a `_dataToArray` function at the start of the file, which does as you suggest and is called throughout. I felt it better to do that rather than replicate the same code multiple times throughout. Since the output of the function will always be a np.array, I don't _think_ we need to wrap them with `cast(Vector, ...)` any more, but I'm still learning about typing.